src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANGE = 100;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static int index;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private String figureType;
+    private Figure figure;
+    private String figureColor;
+
+    public Figure getRandomFigure() {
+        index = random.nextInt(FigureType.values().length);
+        figureType = FigureType.values()[index].name();
+        figureColor = colorSupplier.getRandomColor();
+        switch (figureType) {
```

Comment:
Let's use FigureType enum for this switch case

Reply:
I don't understand what do you mean... So, I put FigureType.values()[index].name() inside SWITCH condition without additional variable (figureType).